### PR TITLE
Use Spring Boot built-in HTTP2 support

### PR DESCRIPTION
**Pull request by cbornet**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

I tested with the `tls`profile activated (very handy btw, thanks @PierreBesson !)
```
curl -v -k --http2 https://localhost:8080/robots.txt
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=; ST=; L=; O=com.mycompany.myapp; OU=Development; CN=Java Hipster
*  start date: Feb 12 21:29:21 2019 GMT
*  expire date: Nov 26 21:29:21 2292 GMT
*  issuer: C=; ST=; L=; O=com.mycompany.myapp; OU=Development; CN=Java Hipster
*  SSL certificate verify result: self signed certificate (18), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x563c43c15900)
> GET /robots.txt HTTP/2
> Host: localhost:8080
> User-Agent: curl/7.58.0
> Accept: */*
> 
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 200 
< expires: 0
< cache-control: no-cache, no-store, max-age=0, must-revalidate
< x-xss-protection: 1 ; mode=block
< pragma: no-cache
< referrer-policy: no-referrer
< accept-ranges: bytes
< date: Tue, 19 Feb 2019 22:11:14 GMT
< last-modified: Tue, 12 Feb 2019 23:49:24 GMT
< strict-transport-security: max-age=31536000 ; includeSubDomains
< x-content-type-options: nosniff
< content-length: 239
< content-type: text/plain
< 
# robotstxt.org/

User-agent: *
Disallow: /api/account
Disallow: /api/account/change-password
Disallow: /api/account/sessions
Disallow: /api/audits/
Disallow: /api/logs/
Disallow: /api/users/
Disallow: /management/
Disallow: /v2/api-docs/
* Connection #0 to host localhost left intact
```

cc @mraible @PierreBesson

### CODE REVIEW

**Comment by DanielFran** [comment]
@cbornet we should delete the jhipster properties also since they are not used now...

**Comment by mraible** [comment]
@DanielFran If we do that, we might want to modify the upgrade generator to remove them too.

**Comment by cbornet** [comment]
@DanielFran indeed !

**Comment by DanielFran** [comment]
@cbornet It is needed also to review documentation like in:
- https://www.jhipster.tech/common-application-properties/#2
- https://www.jhipster.tech/production/#performance

@PierreBesson With implementation of TLS, it seems that the use of property useUndertowUserCipherSuitesOrder disapear. Can you confirm?
´´´java
#Force the server cipher suite to follow the exact order specifying in server.ssl.ciphers (For perfect forward secrecy)
useUndertowUserCipherSuitesOrder: true
´´´

File: generators/server/templates/src/main/resources/config/application-tls.yml.ejs [hunk 1]
```diff
@@ -15,6 +15,5 @@ server:
         key-alias: selfsigned
         ciphers: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384, TLS_DHE_RSA_WITH_AES_128_GCM_SHA256, TLS_DHE_RSA_WITH_AES_256_GCM_SHA384, TLS_DHE_RSA_WITH_AES_128_CBC_SHA, TLS_DHE_RSA_WITH_AES_256_CBC_SHA, TLS_DHE_RSA_WITH_AES_128_CBC_SHA256, TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
         enabled-protocols: TLSv1.2
-jhipster:
-    http:
-        version: V_2_0
+    http2:
```
Comment (PierreBesson) [inline generators/server/templates/src/main/resources/config/application-tls.yml.ejs]: @cbornet I think this should be `server.http2.enabled` and not `jhipster.http2.enabled` !

**Comment by PierreBesson** [comment]
@DanielFran Yes I think this should have been removed before.

**Comment by DanielFran** [comment]
@PierreBesson Thanks for the confimation, I will then remove it from jhipster-bom project.

@cbornet Do not forget to remove reference of this variable also... thanks

File: generators/server/templates/src/main/resources/config/application-tls.yml.ejs (hunk 1 above)
Comment (PierreBesson) [reply generators/server/templates/src/main/resources/config/application-tls.yml.ejs]: Oh my bad, I wasn't reading the diff correctly. Indeed you have removed the jhipster specific property.

**Comment by DanielFran** [comment]
@PierreBesson The property is still used in https://github.com/jhipster/jhipster/blob/master/jhipster-framework/src/main/java/io/github/jhipster/security/ssl/UndertowSSLConfiguration.java

I am not sure if this configuration file is still used?
Should I removed it or only consider that the condition (https://github.com/jhipster/jhipster/blob/master/jhipster-framework/src/main/java/io/github/jhipster/security/ssl/UndertowSSLConfiguration.java#L66) can be removed since it was always true by default?

**Comment by PierreBesson** [comment]
Oh good point ! I guess we should make this the default as this is what you want this all the time. So let's  remove the property.

**Comment by DanielFran** [comment]
@PierreBesson OK, then the PR https://github.com/jhipster/jhipster/pull/196 is done!

**Comment by geyuqiu** [comment]
> @cbornet I take care of removing jhipster properties, see [jhipster/jhipster#194](https://github.com/jhipster/jhipster/pull/194).
> 
> It is needed also to review documentation like in:
> 
> * https://www.jhipster.tech/common-application-properties/#2
> * https://www.jhipster.tech/production/#performance
> * https://www.jhipster.tech/production/#security
> 
> Can you take care of it, please?
> 
> @mraible I did not found anything to add to upgrade generator since those properties are actually in jhipster-bom project.
> Can you clarify, please?
> 
> @PierreBesson With implementation of TLS, it seems that the use of property useUndertowUserCipherSuitesOrder disapear. Can you confirm?
> 
> ```java
> #Force the server cipher suite to follow the exact order specifying in server.ssl.ciphers (For perfect forward secrecy)
> useUndertowUserCipherSuitesOrder: true
> ```

is it done? because useUndertowUserCipherSuitesOrder is commented out in prod yml for jhipster v6.5.x

**Comment by PierreBesson** [comment]
@geyuqiu 
Please don't comment on an **old PR**.
If you think this issue still applies, please [create a new ticket](https://github.com/jhipster/generator-jhipster/issues/new) with proper details.

**Pull request merged**